src/diagnostics/land_compute_methods.jl

```diff
@@ -0,0 +1,146 @@
+@land_compute "vcmax25" SoilCanopyModel p.canopy.photosynthesis.vcmax25
+
+# Canopy - Radiative Transfer
+@land_compute "near_infrared_radiation" SoilCanopyModel p.canopy.radiative_transfer.nir
+@land_compute "near_infrared_radiation_absorbed" SoilCanopyModel p.canopy.radiative_transfer.anir
+@land_compute "near_infrared_radiation_reflected" SoilCanopyModel p.canopy.radiative_transfer.rnir
+@land_compute "near_infrared_radiation_transmitted" SoilCanopyModel p.canopy.radiative_transfer.tnir
+@land_compute "photosynthetically_active_radiation" SoilCanopyModel p.canopy.radiative_transfer.par
+@land_compute "photosynthetically_active_radiation_absorbed" SoilCanopyModel p.canopy.radiative_transfer.apar
+@land_compute "photosynthetically_active_radiation_reflected" SoilCanopyModel p.canopy.radiative_transfer.rpar
+@land_compute "photosynthetically_active_radiation_transmitted" SoilCanopyModel p.canopy.radiative_transfer.tpar
+@land_compute "radiation_longwave_net" SoilCanopyModel p.canopy.radiative_transfer.LW_n
+@land_compute "radiation_shortwave_net" SoilCanopyModel p.canopy.radiative_transfer.SW_n
+
+# Canopy - Transpiration
+@land_compute "canopy_transpiration" SoilCanopyModel p.canopy.transpiration
+
+## Drivers Module ##
+
+@land_compute "air_co2" SoilCanopyModel p.drivers.c_co2
+@land_compute "pressure" SoilCanopyModel p.drivers.P
+@land_compute "rainfall" SoilCanopyModel p.drivers.P_liq
+@land_compute "radiation_longwave_down" SoilCanopyModel p.drivers.LW_d
+@land_compute "radiation_shortwave_down" SoilCanopyModel p.drivers.SW_d
+@land_compute "snowfall" SoilCanopyModel p.drivers.P_snow
+@land_compute "solar_zenith_angle" SoilCanopyModel p.drivers.θs
+@land_compute "specific_humidity" SoilCanopyModel p.drivers.q
+@land_compute "wind_speed" SoilCanopyModel p.drivers.u
+
+## Soil Module ##
+
+@land_compute "infiltration" SoilCanopyModel p.soil.infiltration
+@land_compute "soil_hydraulic_conductivity" SoilCanopyModel p.soil.K
+@land_compute "soil_thermal_conductivity" SoilCanopyModel p.soil.κ
+@land_compute "soil_water_liquid" SoilCanopyModel top_center_to_surface(
+    p.soil.θ_l,
+)
+@land_compute "soil_water_potential" SoilCanopyModel p.soil.ψ
+@land_compute "soil_net_radiation" SoilCanopyModel p.soil.R_n
+@land_compute "soil_temperature" SoilCanopyModel p.soil.T
+
+# Soil - Turbulent Fluxes
+@land_compute "soil_aerodynamic_resistance" SoilCanopyModel p.soil.turbulent_fluxes.r_ae
+@land_compute "soil_latent_heat_flux" SoilCanopyModel p.soil.turbulent_flux.lhf
+@land_compute "soil_sensible_heat_flux" SoilCanopyModel p.soil.turbulent_fluxes.shf
+@land_compute "vapor_flux" SoilCanopyModel p.soil.turbulent_fluxes.vapor_flux
+
+# Soil - SoilCO2
+@land_compute "heterotrophic_respiration" SoilCanopyModel p.soilco2.top_bc
+@land_compute "soilco2_diffusivity" SoilCanopyModel p.soilco2.D
+@land_compute "soilco2_source_microbe" SoilCanopyModel p.soilco2.Sm
+
+# variables stored in Y (prognostic or state variables)
```

Comment:
Do we really only have 3? 
Surely not. for example, it seems like soil moisture is missing below.

Reply:
Y.soil.\rhoe_int
Y.soil.\vartheta_l
Y.soil.\theta_i
Y.soilco2.C
(only four for soil)

canopy T and canopy water content are prognostic

Reply:
Is soil temperature prognostic? 
Or is it diagnostic, calculated from rhoe_int?

Reply:
Temperature is diagnostic and computed from all three prognostic variables